ui/src/components/tag/TagList.js

```diff
@@ -0,0 +1,483 @@
+                }),
+            MODAL_TIME_OUT
+        );
+    }
+
+    closeModal() {
+        this.setState({
+            successMessage: '',
+            showSuccess: false,
+        });
+    }
+
+    render() {
+        let addNewTag = this.addNewTag.bind(this);
+        let validateTagExist = this.validateTagExist.bind(this);
+        const left = 'left';
+        const center = 'center';
+        let rows = '';
+        if (this.state.tags) {
+            const clonedTags = JSON.parse(JSON.stringify(this.state.tags));
+            rows = Object.entries(clonedTags).map((item, i) => {
+                const tagKey = item[0];
+                const tagValues = item[1];
+                let onClickDeleteTag = this.onClickDeleteTag.bind(this, tagKey);
+                let onClickDeleteTagValue = this.onClickDeleteTagValue.bind(
+                    this
+                );
+                let onClickEditTag = this.onClickEditTag.bind(this);
+                let color = '';
+                if (i % 2 === 0) {
+                    color = colors.row;
+                }
+                let toReturn = [];
+                toReturn.push(
+                    <TagRow
+                        key={tagKey}
+                        tagKey={tagKey}
+                        tagValues={tagValues}
+                        color={color}
+                        api={this.api}
+                        _csrf={this.props._csrf}
+                        onClickDeleteTag={onClickDeleteTag}
+                        onClickDeleteTagValue={onClickDeleteTagValue}
+                        onClickEditTag={onClickEditTag}
+                    />
+                );
+                return toReturn;
+            });
+        }
+        let addTag = this.state.showAddTag ? (
+            <AddTag
+                showAddTag={this.state.showAddTag}
+                editMode={this.state.editMode}
+                onCancel={this.closeAddTag}
+                onSubmit={this.reloadTags}
+                resource={
+                    this.state.category === 'domain'
+                        ? this.props.domain
+                        : this.props.role
+                }
+                api={this.api}
+                _csrf={this.props._csrf}
+                addNewTag={addNewTag}
+                editedTagKey={this.state.editedTagKey}
+                editedTagValues={this.state.editedTagValues}
+                errorMessage={this.state.errorMessage}
+                validateTagExist={validateTagExist}
+            />
+        ) : (
+            ''
+        );
+        return (
+            <TagsSectionDiv data-testid='tag-list'>
+                <AddContainerDiv>
+                    <div>
+                        <Button secondary onClick={this.openAddTag}>
+                            Add Tag
+                        </Button>
+                        {addTag}
+                    </div>
+                </AddContainerDiv>
+                <TagTable>
```

Comment:
display the table only if there are tags present otherwise show a message like - Click on Add Tag to create a new tag